### ISSUE TITLE
Preventing JavaScript URLs upon parsing markdown links

### ISSUE THREAD

**Issue by CrimsonCodes0** [rsms/markdown-wasm#14]
I was looking at using this project for a server-side Markdown parser, but someone I was working with tried out the demo and pointed out that this markdown:
```markdown
[some text](javascript:alert("xss"))
```
renders as an actual JavaScript link. Does markdown-wasm offer more fine-grained control to prevent this specific URL type from being parsed?

If not, we can attempt to remove them manually (using JavaScript), but it would be great if it could directly be dealt with in the Wasm parser itself.

**Comment by lostfictions** [comment]
Looks like that's considered a valid link under the CommonMark spec, which md4c (and thus markdown-wasm) strictly comply to: https://spec.commonmark.org/0.29/#link-destination

I'd recommend just passing output through [DOMPurify](https://github.com/cure53/DOMPurify).

**Comment by 00ff0000red** [comment]
In my particular case, javascript: links do nothing on my site, since our server sends Content-Security-Policy HTTP headers that strictly deny inline JavaScript and CSS, so I didn't even need to modify the output.

But yes, it is a valid URL, that I have no problem with, but users might appreciate a flag that prevents emitting them.

**Comment by rsms** [comment]
Ah, this would definitely be a nice option to add, to allow this as an opt-in (disallowed by default.)
For now, a post-processing step on the generated HTML is a good idea if you allow untrusted users to enter & render HTML.

**Comment by rsms** [comment]
4b48783c3e209d900a459a2bd76aa314cb62f7e1 (released with v1.2.0 on NPM) includes a fix for this. Starting with this version, by default any href starting with "javascript:" is now stripped. "javascript:" URIs can be explicitly enabled by setting the option `allowJSURIs`.

Example input:

```md
[XSS test](javAscRipt:alert("xss"))
```

Output:

```
<p><a href="">XSS test</a></p>
```
